src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -920,7 +942,18 @@ def _complete_osd_upgrade(self, target_major: str, target_major_name: str) -> No
 
     def _complete_mds_upgrade(self) -> None:
         assert self.upgrade_state is not None
-        if self.upgrade_state.fs_original_max_mds:
+        if self.upgrade_state.fail_fs:
+            for fs in self.mgr.get("fs_map")['filesystems']:
+                fs_name = fs['mdsmap']['fs_name']
+                ret, _, err = self.mgr.check_mon_command({
+                    'prefix': 'fs set',
+                    'fs_name': fs_name,
+                    'var': 'joinable',
+                    'val': 'true',
+                })
+
+        if self.upgrade_state.fs_original_max_mds and \
```

Comment:
can we just use an `elif` here rather than rechecking `fail_fs`?

Reply:
Yeah It can be done, I'll change it, thanks!

Reply:
Done